metron-interface/metron-rest/src/test/java/org/apache/metron/rest/controller/AlertsProfileControllerIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user1ProfileJson + "]"));
+
+    this.mockMvc.perform(post(url).with(httpBasic(user2, password)).with(csrf())
+        .contentType(MediaType.parseMediaType("application/json;charset=UTF-8"))
+        .content(user2ProfileJson))
+        .andExpect(status().isCreated())
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user1ProfileJson));
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user1ProfileJson + "," + user2ProfileJson + "]"));
+
+    this.mockMvc.perform(delete(url + "/user1").with(httpBasic(admin, password)))
+        .andExpect(status().isOk());
+
+    this.mockMvc.perform(delete(url + "/user1").with(httpBasic(admin, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user2ProfileJson + "]"));
+
+    this.mockMvc.perform(delete(url + "/user2").with(httpBasic(admin, password)))
+        .andExpect(status().isOk());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
```

Comment:
There is a technical issue here and that's annotating multiple methods with @Test.  This changes the lifecycle of the tests and makes it so that each @Test method has to do it's own setup and teardown.  Not a big deal with unit tests because functions are tested in isolation with mock objects but it's harder for integration tests.  This test is not mocking anything, the same in-memory components that our other integration tests use serve as the backend.  We definitely don't want to spin them up/down before each test.  In fact, the REST integration tests only spin them up/down once for all tests.

If we were to organize the cases into different methods but NOT annotate them with @Test would that be ok?  I would rather document the state each test expects rather than recreating it.

Reply:
Sounds reasonable. That would definitely be an improvement. If there are no mocks here, we should probably change this `this.mockMvc...` to this `this.mvc...` That is misleading.

Reply:
I added more documentation and split the use cases into separate methods.  Let me know if this is clear enough now.

My previous comment wasn't entirely accurate.  The test is mocking the http layer which is what the mockMvc object is for.  The backend is not mocked though and my comments about  not wanting to initialize backend components before each test still stands.